ganttproject/src/main/java/net/sourceforge/ganttproject/storage/ProjectDatabase.kt

```diff
@@ -20,9 +20,12 @@ along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
 package net.sourceforge.ganttproject.storage
 
 import net.sourceforge.ganttproject.task.Task
+import net.sourceforge.ganttproject.task.dependency.TaskDependency
 import kotlin.jvm.Throws
 
-open class ProjectDatabaseException(message: String): Exception(message)
+open class ProjectDatabaseException(message: String): Exception(message) {
+  constructor(message: String, cause: Exception): this("$message. Caused by ${cause.message}")
```

Comment:
There is a super-constructor with the cause argument. It creates a chain of exceptions and prints the stack traces of all exceptions in the chain.